### PR TITLE
Remove Non-functional Quorum Replication Code

### PR DESCRIPTION
**Pull request by Maithem**
## Overview
Remove Non-functional Quorum Replication Code

Why should this be merged: Doing some major clean ups and don't want to apply those changes to non-function code. 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: test/src/test/java/org/corfudb/runtime/clients/LogUnitHandlerTest.java [hunk 1]
```diff
@@ -263,93 +262,17 @@ public void flushLogUnitCache() throws Exception {
...
-            assertThat(log.getType()).isEqualTo(DataType.DATA);
-            assertThat(log.getPayload(new CorfuRuntime())).isEqualTo(testString);;
-        }
-        r = client.read(0).get().getAddresses().get(0L);
-        assertThat(r.getType()).isEqualTo(DataType.DATA);
-        assertThat(r.getPayload(new CorfuRuntime())).isEqualTo(testString);
-    }
 
-    private ILogData.SerializationHandle createEmptyData(long position, DataType type, IMetadata.DataRank rank) {
+    private ILogData.SerializationHandle createEmptyData(long position, DataType type) {
```
Comment (annym) [inline test/src/test/java/org/corfudb/runtime/clients/LogUnitHandlerTest.java]: Codacy Issue found: [Avoid unused private methods such as 'createEmptyData(long,DataType)'.](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4704713)

**Comment by annym** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           

Complexity decreasing per file
==============================
+ runtime/src/main/java/org/corfudb/protocols/wireprotocol/IMetadata.java  -1
+ infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java  -2
         

Clones removed
==============
+ runtime/src/main/java/org/corfudb/runtime/view/Layout.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4704713&bid=15717525)

**Comment by annym** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Complexity decreasing per file
==============================
+ runtime/src/main/java/org/corfudb/protocols/wireprotocol/IMetadata.java  -1
+ infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java  -2
         

Clones removed
==============
+ runtime/src/main/java/org/corfudb/runtime/view/Layout.java  -2
+ test/src/test/java/org/corfudb/infrastructure/LogUnitServerTest.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4704713&bid=15717525)

**Comment by annym** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Complexity decreasing per file
==============================
+ infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java  -2
+ runtime/src/main/java/org/corfudb/protocols/wireprotocol/IMetadata.java  -1
         

Clones removed
==============
+ test/src/test/java/org/corfudb/infrastructure/LogUnitServerTest.java  -2
+ runtime/src/main/java/org/corfudb/runtime/view/Layout.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4704713&bid=15717525)

File: test/src/test/java/org/corfudb/infrastructure/LayoutBuilderTest.java [hunk 2]
```diff
@@ -92,6 +91,7 @@ public void checkRemovalOfNodes() throws LayoutModificationException {
         failedNodes.clear();
         failedNodes.add(allNodes.get(0));
         failedNodes.add(allNodes.get(2));
+        failedNodes.add(allNodes.get(3));
```
Comment (zhangn49) [inline test/src/test/java/org/corfudb/infrastructure/LayoutBuilderTest.java]: Travis is failing due to magic number 3.

Comment (Maithem) [reply test/src/test/java/org/corfudb/infrastructure/LayoutBuilderTest.java]: Addressed. Thank.s

File: format/proto/types.proto [hunk 3]
```diff
@@ -14,20 +13,13 @@ enum CheckpointEntryType {
...
-    required int64 uuid_least_significant = 3;
-}
-
 message LogEntry {
     optional DataType data_type = 1;
     optional bytes data = 2;
     optional int64 global_address = 3;
     repeated string	streams = 6;
     map<string, int64> logical_addresses = 7;
     map<string, int64> backpointers = 8;
```
Comment (Maithem) [inline format/proto/types.proto:22]: In order to avoid log migration, we need to deprecate this and not remove it.

**Pull request closed without merging**